phpBB/includes/acp/acp_storage.php

```diff
@@ -96,8 +122,173 @@ public function overview($id, $mode)
 		// Set page title
 		$this->page_title = 'STORAGE_TITLE';
 
+		$action = $this->request->variable('action', '');
+		$this->load_state();
+
+		// If user cancelled to continue, remove state
+		if ($this->request->is_set_post('cancel', false))
+		{
+			if (!check_form_key($form_key) || !check_link_hash($this->request->variable('hash', ''), 'acp_storage'))
+			{
+				trigger_error($this->lang->lang('FORM_INVALID') . adm_back_link($this->u_action), E_USER_WARNING);
+			}
+
+			if ($this->request->variable('cancel', false))
+			{
+				$action = '';
+				$this->state = false;
+				$this->save_state();
+			}
+		}
+
+		if ($action)
+		{
+			switch ($action)
```

Comment:
I don't think it make sense to have a switch here. This would be shorter and reduce the complexity:
```
if ($action == 'progress_bar')
{
	$this->display_progress_bar();
}
else if ($action != 'update')
{
	trigger_error('NO_ACTION', E_USER_ERROR);
}
```

Reply:
This is almost the same code than in acp_search, i didn't changed it much to have both files similar, so if in the future is changed there arent 10 diferent ways to do the same.